### PR TITLE
Remove support for old test case files

### LINKED ISSUE #97: Drop support for legacy testcase file format
We've supported the legacy testcase file format for quite some time now and the readme file has included a jq command for converting between the formats so let's drop the old format in the 2.0 release.

### PR DESCRIPTION
**Pull request by magnusbaeck**
Originally, the "input" and "expected" keys were at the top level of the test case files, and since the introduction of the "testcases" key both have been supported. Keeping the old keys is only confusing, so now that we're stepping the major version we should seize the opportunity to drop support for the old format.

Rather than just drop them entirely and let the YAML package report the error, we capture any values given these keys and provide the user with a useful error message.

Fixes #97

### CODE REVIEW

File: internal/testcase/testcase_test.go [hunk 1]
```diff
@@ -126,14 +116,25 @@ func TestNew(t *testing.T) {
...
 		tcs, err := New(bytes.NewReader([]byte(c.input)), "json")
-		if err != nil {
-			t.Errorf("Test %d: %q input: %s", i, c.input, err)
-			break
-		}
-		resultJSON := marshalTestCaseSet(t, tcs)
-		expectedJSON := marshalTestCaseSet(t, &c.expected)
-		if expectedJSON != resultJSON {
-			t.Errorf("Test %d:\nExpected:\n%s\nGot:\n%s", i, expectedJSON, resultJSON)
+		if c.expectedError != "" {
```
Comment (breml) [inline internal/testcase/testcase_test.go:119]: Me personally, I don't like such conditions in the test code. In this case, I would consider to move these cases into a separate test function.

File: internal/testcase/testcase.go [hunk 2]
```diff
@@ -199,6 +203,16 @@ func New(reader io.Reader, configType string) (*TestCaseSet, error) {
...
 
+	// Bail out if the old testcase file format is used.
+	if len(tcs.LegacyExpectedEvents) != 0 {
+		return nil, errors.New(`testcase file contained deprecated "expected" key; ` +
+			`specify an array of testcase objects under the "testcases" key instead`)
+	}
+	if len(tcs.LegacyInputLines) != 0 {
+		return nil, errors.New(`testcase file contained deprecated "input" key; ` +
+			`specify an array of testcase objects under the "testcases" key instead`)
+	}
```
Comment (breml) [inline internal/testcase/testcase.go:214]: I like this approach, since it gives the user a meaningful error.

**Review by breml — APPROVED** [review]
LGTM, I would structure the tests a bit differently, but this is not a blocking issue.

File: internal/testcase/testcase_test.go (hunk 1 above)
Comment (magnusbaeck) [reply internal/testcase/testcase_test.go]: I usually don't have a problem with it since the setup of the test often is significantly more code than the conditional itself, and splitting the test function in two duplicates all the setup code. Unless you extract the setup to a reusable function, but that has its own drawbacks.

However, in this particular case the split makes sense.

**Review by breml — APPROVED** [review]
LGTM
